### PR TITLE
Add notifications drawer

### PR DESCRIPTION
**Pull request by cstns**
## Description

Adds a notifications drawer boilerplate. Currently only supports team invitations but is easily extended to future notifications

## Related Issue(s)
closes https://github.com/FlowFuse/flowfuse/issues/4184

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: frontend/src/components/NotificationsButton.vue
Comment (joepavitt) [inline frontend/src/components/NotificationsButton.vue:48]:
```suggestion
    padding: 18px;
```

File: frontend/src/components/NotificationsButton.vue
Comment (joepavitt) [inline frontend/src/components/NotificationsButton.vue:58]:
```suggestion
      width: 24px;
      height: 24px;
```

**Review by joepavitt — CHANGES_REQUESTED** [review]
- Odd glitch on close where it seems to span the full width for a brief millisecond or two, then pull back?

https://github.com/user-attachments/assets/a6082067-0bcd-4ce3-bd85-6237dd8007d9

- I'd also expect to be able to click the notifications icon _again_ to close too, not just have to click into empty space.
- We should also change the red of the `ff-notification-pill` to `#d12c2b` as it's too dark at the moment
- We also need to improve the padding and vertical alignment of the notifications themselves:

Screenshot 2024-07-18 at 18 20 49

I'm happy to do that myself though if you don't mind handling the other recommendations

**Comment by joepavitt** [comment]
As an FYI too, we have `ff-icon`, `ff-icon-lg` and `ff-icon-sm` which you can use here too. The icon in the notification itself for example can just use `ff-icon ff-icon-lg` and it'll then be 24px squared.

**Comment by cstns** [comment]
I'm looking into the collapse bug! Feel free to make any adjustments you feel necessary!

**Comment by joepavitt** [comment]
Made some structural improvmenets by having all of the structure live inside a `Notification.vue` component with relevant slots that the bespoke notifications can extend.

I've also then modified the CSS slightly to implement the following changes:

### Before

Screenshot 2024-07-19 at 12 36 54

### After

Screenshot 2024-07-19 at 12 34 07

**Comment by joepavitt** [comment]
Waiting for tests to run - but will approve once they're successful

**Review by joepavitt — APPROVED** [review]

**Pull request merged**
